src/main/java/org/commonjava/util/partyline/impl/infinispan/InfinispanJF.java

```diff
@@ -580,7 +580,9 @@ public int read() throws IOException
 
             }
 
-            final int result = block.readFromBuffer();
+            final int result = block.readFromBuffer(); // TODO: isn't the block shared? reading from buffer would change the position.
```

Comment:
Agreed, we need to keep a local ByteBuffer and position in the block so we can read independently from other threads.